purchases/src/main/kotlin/com/revenuecat/purchases/common/OfferingParser.kt

```diff
@@ -49,7 +52,24 @@ internal abstract class OfferingParser {
             }
         }
 
-        return Offerings(offerings[currentOfferingID], offerings)
+        val placements: Placements? = offeringsJson.optJSONObject("placements")?.let {
+            val fallbackOfferingId = it.getNullableString("fallback_offering_id")
+            val offeringIdsByPlacement = it.optJSONObject("offering_ids_by_placement")
+                ?.toMap<String?>()
+                ?.replaceJsonNullWithKotlinNull()
+
+            return@let offeringIdsByPlacement?.let {
+                Placements(
+                    fallbackOfferingId = fallbackOfferingId,
+                    offeringIdsByPlacement = offeringIdsByPlacement,
+                )
+            } ?: run {
+                errorLog(OfferingStrings.PLACEMENTS_EMPTY_ERROR)
```

Comment:
I'm unsure of this situation... will it only happen when there is a placements key but not an offering_ids_by_placement key? I guess that that should never happen but should be ok to log it. Maybe leave a comment to expect this situation to never happen?

Reply:
This should never happen because its not allowed on the backend. I didn't want to _not_ catch it but it could be unclear why things aren't working if it does happen 🤷 

I will leave a comment!